iceoryx_posh/include/iceoryx_posh/internal/popo/untyped_publisher.inl

```diff
@@ -22,25 +22,27 @@ namespace iox
 {
 namespace popo
 {
-template <typename base_publisher_t>
-inline UntypedPublisherImpl<base_publisher_t>::UntypedPublisherImpl(const capro::ServiceDescription& service,
-                                                                    const PublisherOptions& publisherOptions)
+template <typename H, typename base_publisher_t>
```

Comment:
`HeaderType` or `Header_t`? I think we're really inconsistent with naming template params, makes reading harder :disappointed:

Reply:
It's just `H` to be consistent with the typed API. There was already a `T` template parameter and then I added the `H` one

Reply:
Please change the template param in the typed API, too. Same goes for the `base_publisher_t`. I think in iceoryx most of the time we use `FooType`.

Reply:
I can change `base_publisher_t` (in a separate PR together with the typed API) but would keep the `H`.

Reply:
I vote for short dummy variables in the inl, they do not have to be the same as in the hpp (readability). In fact, if they are not used even typename is enough. We need a convention for those _t suffixes but IMO they should not be used in template parameters.